js/kucoinfutures.js

```diff
@@ -1068,28 +1068,20 @@ module.exports = class kucoinfutures extends kucoin {
         const takeProfitPrice = this.safeValue (params, 'takeProfitPrice');
         const isStopLoss = stopLossPrice !== undefined;
         const isTakeProfit = takeProfitPrice !== undefined;
-        const stop = this.safeString (params, 'stop');
-        const stopPriceType = this.safeString (params, 'stopPriceType', 'MP');
         if (stopPrice) {
-            if (stop === undefined) {
-                request['stop'] = (side === 'buy') ? 'down' : 'up';
-            }
+            request['stop'] = (side === 'buy') ? 'up' : 'down';
             request['stopPrice'] = this.priceToPrecision (symbol, stopPrice);
-            request['stopPriceType'] = stopPriceType;
+            request['stopPriceType'] = 'MP';
         } else if (isStopLoss || isTakeProfit) {
             if (isStopLoss) {
-                if (stop === undefined) {
-                    request['stop'] = (side === 'buy') ? 'up' : 'down';
-                }
+                request['stop'] = (side === 'buy') ? 'down' : 'up';
```

Comment:
If I'm creating a stop-loss order for a long position wouldn't the `stopPrice` be triggered from the price going downwards to open a `sell` order at the `price`, and vice versa for takeProfit and short positions

Reply:
> If I'm creating a stop-loss order for a long position wouldn't the `stopPrice` be triggered from the price going `down` to open a `sell` order at the `price`, and vice versa for takeProfit and short positions

for a long **position** your stop order will be a sell, because you're exiting

Reply:
> > > If I'm creating a stop-loss order for a long position wouldn't the `stopPrice` be triggered from the price going `down` to open a `sell` order at the `price`, and vice versa for takeProfit and short positions
> > 
> > 
> > for a long **position** your stop order will be a sell, because you're exiting
> 
> Right so we should probably change the stopLoss and takeProfit stop requests back then

on those ones, the side refers to the direction of the original order, so the buy/sell side of the stopLoss is opposite of the side of the order

Reply:
@samgermain 
I think the `stop` parameter being `up` or `down` is referring to the stop-direction not the direction of the current position. The kucoin api doesn't have true stop-losses or take-profits yet so these are order type 2 from the stopLoss schema #13822.

I tested your approach alongside the kucoin futures website and these are the responses I received from the orders:
### set side to sell:
`kucoinfutures createOrder BTC/USDT:USDT limit sell 1 13000 '{"stopLossPrice":"14000"}'`
`kucoinfutures createOrder BTC/USDT:USDT limit sell 1 14000 '{"stopLossPrice":"13000"}'`
result:
`Sell 1 BTC Perpetual/USDT contract(s) at 13000 . The order has been canceled. Caused by: Order price cannot be lower than 21465.611`

### set side to buy:
`kucoinfutures createOrder BTC/USDT:USDT limit buy 1 13000 '{"stopLossPrice":"14000"}'`
`kucoinfutures createOrder BTC/USDT:USDT limit buy 1 14000 '{"stopLossPrice":"13000"}'`
result:
`creates a buy order`

### current approach
`kucoinfutures createOrder BTC/USDT:USDT limit sell 1 13000 '{"stopLossPrice":"14000"}'`
result:
`creates an emulated stop-loss sell order with triggerPrice 14000 and order price 13000`

Reply:
> The `stopLossPrice` can't exist without another order though, so when your doing
> 
> `kucoinfutures createOrder BTC/USDT:USDT limit sell 1 13000 '{"stopLossPrice":"14000"}'`
> 
> I thought that you would be creating a limit order to sell 1 BTC/USDT contract 13000, and then if this limit order is hit, the position will have a stop loss price of 14000

I thought that would be the best way to handle stop-losses initially as well but it's actually kind of nice doing it this way and sort of emulating stop-losses using stop orders because a lot of exchange API's don't support order type 3 yet